### PR TITLE
{Docs} Quote code in help messages with backticks

### PR DESCRIPTION
**Pull request by jiasli**
**Description**  
Fix https://github.com/Azure/azure-cli/issues/10698: `az ad app permission add` description not understandable

## Symptom

The doc for `az ad app permission add` is not rendered correctly, with the value of `--api-permissions` missing. 

https://docs.microsoft.com/en-us/cli/azure/ad/app/permission

image

## Root Cause

The reason is that the doc generator parses `<resource-access-id>` and `<type>` as a possible HTML entities:

https://github.com/Azure/azure-cli/blob/1f34245e83fbb68777c26a764f9cb5e8b86c0051/src/azure-cli/azure/cli/command_modules/role/_params.py#L61

but they don't match up with any entities on the [Docs HTML Whitelist](https://review.docs.microsoft.com/en-us/help/onboard/admin/html-whitelist?branch=master), and the content is discarded. 

**This happens not only to this command but all over the place, including VM, Network and more commands.** 

##  Solution

This PR quotes the text with backticks `` ` `` like what `azure-cli-core` does for `--resource-group`: https://github.com/Azure/azure-cli/blob/c4e91c00b7266efb7fb05dcd75870ea2dde713cc/src/azure-cli-core/azure/cli/core/commands/parameters.py#L241

In this way, the doc is rendered correctly (even better with highlighting due to backtick (`` ` ``) being interpreted as `code` in markdown), such as 

https://docs.microsoft.com/en-us/cli/azure/acs

image

⚠ But the disadvantage is the `--help` message may look verbose. 

## Remarks

I searched the occurrence of `<>` with regex `<.+?>` (`?` stands for non-greedy match) and added backticks manually.

### CODE REVIEW

File: src/azure-cli/azure/cli/command_modules/keyvault/_params.py [hunk 1]
```diff
@@ -93,7 +93,7 @@ class CLIJsonWebKeyOperation(str, Enum):
     with self.argument_context('keyvault create', arg_group='Network Rule') as c:
         c.argument('network_acls', type=validate_file_or_dict,
                    help='Network ACLs. It accepts a JSON filename or a JSON string. JSON format: '
-                        '{\\"ip\\":[<ip1>, <ip2>...],\\"vnet\\":[<vnet_name_1>/<subnet_name_1>,<subnet_id2>...]}')
+                        '`{\\"ip\\":[<ip1>, <ip2>...],\\"vnet\\":[<vnet_name_1>/<subnet_name_1>,<subnet_id2>...]}`')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/keyvault/_params.py:96]: @bim-msft

File: src/azure-cli/azure/cli/command_modules/resource/_params.py [hunk 2]
```diff
@@ -391,7 +391,7 @@ def load_arguments(self, _):
 
     with self.argument_context('managedapp definition create') as c:
         c.argument('lock_level', arg_type=get_enum_type(ApplicationLockLevel), help='The type of lock restriction.')
-        c.argument('authorizations', options_list=['--authorizations', '-a'], nargs='+', help="space-separated authorization pairs in a format of <principalId>:<roleDefinitionId>")
+        c.argument('authorizations', options_list=['--authorizations', '-a'], nargs='+', help="space-separated authorization pairs in a format of `<principalId>:<roleDefinitionId>`")
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/resource/_params.py:394]: @Juliehzl @zhoxing-ms

File: src/azure-cli/azure/cli/command_modules/network/_params.py [hunk 3]
```diff
@@ -1460,7 +1460,7 @@ def load_arguments(self, _):
 
     with self.argument_context('network public-ip prefix') as c:
         c.argument('public_ip_prefix_name', name_arg_type, completer=get_resource_name_completion_list('Microsoft.Network/publicIPPrefixes'), id_part='name', help='The name of the public IP prefix.')
-        c.argument('prefix_length', options_list='--length', help='Length of the prefix (i.e. XX.XX.XX.XX/<Length>)')
+        c.argument('prefix_length', options_list='--length', help='Length of the prefix (i.e. `XX.XX.XX.XX/<Length>`)')
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/network/_params.py:1463]: @MyronFanQiu

File: src/azure-cli/azure/cli/command_modules/vm/_params.py [hunk 4]
```diff
@@ -541,7 +541,7 @@ def load_arguments(self, _):
 
     with self.argument_context('vmss create', min_api='2017-03-30', arg_group='Network') as c:
         c.argument('public_ip_per_vm', action='store_true', help="Each VM instance will have a public ip. For security, you can use '--nsg' to apply appropriate rules")
-        c.argument('vm_domain_name', help="domain name of VM instances, once configured, the FQDN is 'vm<vm-index>.<vm-domain-name>.<..rest..>'")
+        c.argument('vm_domain_name', help="domain name of VM instances, once configured, the FQDN is `vm<vm-index>.<vm-domain-name>.<..rest..>`")
```
Comment (jiasli) [inline src/azure-cli/azure/cli/command_modules/vm/_params.py:544]: @qwordy

**Comment by qwordy** [comment]
How do you preview the rendered html page to see the result is expected.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 4 above)
Comment (qwordy) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:544]: How do you send picture?

File: src/azure-cli/azure/cli/command_modules/keyvault/_params.py (hunk 1 above)
Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/keyvault/_params.py:96]: Thanks for pointing it out.

File: src/azure-cli/azure/cli/command_modules/vm/_params.py (hunk 4 above)
Comment (bim-msft) [reply src/azure-cli/azure/cli/command_modules/vm/_params.py:544]: > How do you send picture?

Create a screenshot and paste it into input box? I guess.
This is the way I used.

**Comment by jiasli** [comment]
> How do you preview the rendered html page to see the result is expected?

I don't know how to "preview the rendered html page". I am deducing it from the global params `--resource-group` and `--location`. +@daxianji007 to clarify. 

> How do you open a draft PR?

See [this blog](https://github.blog/2019-02-14-introducing-draft-pull-requests/) and [this help](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/creating-a-pull-request#creating-the-pull-request):

**Comment by qwordy** [comment]
I encounter a same issue of a new parameter I add in Azure CLI 2.4.0. Only at release time can I see the HTML page. This is the problem!

**Comment by yonzhan** [comment]
add to S169

File: doc/authoring_help.md [hunk 5]
```diff
@@ -81,6 +81,7 @@ helps['account'] = """
 - Don't use "etc". Sometimes it makes sense to spell out a list completely. Sometimes it works to say "like ..." instead of "..., etc".
 - Use active voice. For example, say "Update web app configurations" instead of "Updates web app congfigurations" or "Updating web app configurations".
 - Don't use highly formal language. If you imagine that another dev sat down with you and you were telling him what he needs to know to use the command, that's exactly what you need to write, in those words.
+- If the help message contains **angle brackets**, like `<name>`, it will be parsed as an HTML tag during document rendering. To bypass that, quote the content with backticks `` `<name>` `` to tell the document renderer to parse it as **code**.
```
Comment (jiasli) [inline doc/authoring_help.md:84]: > Can we put a guidance in [authoring_help.md](https://github.com/Azure/azure-cli/blob/dev/doc/authoring_help.md)?

Good point, added.

File: doc/authoring_help.md [hunk 6]
```diff
@@ -26,9 +26,9 @@ To override help for a given command:
 >  2. The Help Authoring System now supports **help.yaml** files. Eventually, **_help.py** files will be replaced by **help.yaml**.
 
 
-### Example YAML help file, _help.py ###
+### Example YAML help file, _help.py
```
Comment (jiasli) [inline doc/authoring_help.md:29]: `_` breaks GitHub's highlighting.

image
```suggestion
### Example YAML help file, \_help.py
```

**Pull request merged**